src/reducers/total_transactions_count_by_contract_addresses.rs

```diff
@@ -0,0 +1,140 @@
+use gasket::{
+    runtime::{spawn_stage, WorkOutcome},
+};
+use pallas::{ledger::primitives::alonzo};
+use serde::Deserialize;
+
+use crate::{bootstrap, crosscut, model};
+
+type InputPort = gasket::messaging::InputPort<model::ChainSyncCommandEx>;
+type OutputPort = gasket::messaging::OutputPort<model::CRDTCommand>;
+
+#[derive(Deserialize)]
+pub struct Config {
+    pub key_prefix: Option<String>,
+}
+
+pub struct Worker {
+    config: Config,
+    input: InputPort,
+    output: OutputPort,
+    ops_count: gasket::metrics::Counter,
+}
+
+impl Worker {
+
+
+    fn increment_for_contract_address(
+        &mut self,
+    ) -> Result<(), gasket::error::Error> {
+        let key = match &self.config.key_prefix {
+            Some(prefix) => prefix.to_string(),
+            None => "total_transactions_count_by_contract_addresses".to_string()
+        };
+    
+        let crdt = model::CRDTCommand::PNCounter(key, "1".to_string());
+        self.output.send(gasket::messaging::Message::from(crdt))?;
+        self.ops_count.inc(1);
+
+        Ok(())
+    }
+
+    fn reduce_alonzo_compatible_tx(
+        &mut self,
+        tx: &alonzo::TransactionBody,
+    ) -> Result<(), gasket::error::Error> {
+        let is_smart_contract_transaction = tx.iter()
```

Comment:
@scarmuega is there any more clever way to check if a transaction is in fact smart contract transaction? 

Right now I am checking if any of the output addresses is in fact smart contract type of address, there could be a more clever way I guess..

Reply:
I found this:
```
pub enum TransactionBodyComponent {
    Inputs(MaybeIndefArray<TransactionInput>),
    Outputs(MaybeIndefArray<TransactionOutput>),
    Fee(u64),
    Ttl(u64),
    Certificates(MaybeIndefArray<Certificate>),
    Withdrawals(KeyValuePairs<RewardAccount, Coin>),
    Update(Update),
    AuxiliaryDataHash(ByteVec),
    ValidityIntervalStart(u64),
    Mint(Multiasset<i64>),
    ScriptDataHash(Hash<32>),
    Collateral(MaybeIndefArray<TransactionInput>),
    RequiredSigners(MaybeIndefArray<AddrKeyhash>),
    NetworkId(NetworkId),
}
```

ScriptDataHash -> will this be empty if this is not smart contract transaction?

Reply:
Depends on what you mean by "Smart contract transaction". If you are looking for transactions that execute scripts, then you need to look at input-related components, since script execution only takes place when you're consuming a UTxO that's sitting on a script address. A way of doing this is checking for redeemers in the witness set.